### PR TITLE
8065422: Trailing dot in hostname causes TLS handshake to fail with SNI disabled

### PR DESCRIPTION
**Pull request by wkia**
_Almost_ clean backport.

JDK8 has the same issue with processing host names for SNI. It'd be good to keep JDK8 up to date with the upstream. No risks.

There were 2 changes with test files paths:

- Original `test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java` added and moved to `jdk/test/javax/net/ssl/ServerName/EndingDotHostname.java`.
- Original `test/jdk/javax/net/ssl/templates/SSLExampleCert.java` added and moved to `jdk/test/javax/net/ssl/templates/SSLExampleCert.java`.

And the original changes use JDK17 strings in `SSLExampleCert.java`, so it was adapted for JDK8 withthe additional commit as it was done for JDK11 backport.

Please note that new test added by the change fails for now because of the certs are expired in March, 2022. To fix this, there are additional JDK fixes being backported as dependant PRs

- #206 
- #207

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8065422](https://bugs.openjdk.org/browse/JDK-8065422): Trailing dot in hostname causes TLS handshake to fail with SNI disabled

### Reviewers
 * [Yuri Nesterenko](https://openjdk.org/census#yan) (@yan-too - **Reviewer**) ⚠️ Review applies to [84cd2b55](https://git.openjdk.org/jdk8u-dev/pull/205/files/84cd2b557e3ec372a898ff3d93cadbae0c590f07)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**) ⚠️ Review applies to [84cd2b55](https://git.openjdk.org/jdk8u-dev/pull/205/files/84cd2b557e3ec372a898ff3d93cadbae0c590f07)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/205/head:pull/205` \
`$ git checkout pull/205`

Update a local copy of the PR: \
`$ git checkout pull/205` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/205/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 205`

View PR using the GUI difftool: \
`$ git pr show -t 205`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/205.diff">https://git.openjdk.org/jdk8u-dev/pull/205.diff</a>

</details>

### CODE REVIEW

**Review by gnu-andrew — APPROVED** [review]
Clean backport, simple bug fix.

**Comment by gnu-andrew** [comment]
This has a `jdk8u-fix-yes` approval already. Please run the `integrate` command and then one of us can sponsor this to get it in.

**Comment by wkia** [comment]
The original changes use JDK17 strings in `SSLExampleCert.java`, so it was adapted for JDK8 withthe additional commit as it was done for JDK11 backport.

**Comment by phohensee** [comment]
Pre-submit test failures look like an infrastructure issue. @wkia, the backport has been approved, so if you add a "/integrate" comment, I or another committer can sponsor it.

**Pull request closed without merging**
